docs/conandata_yml_format.md

```diff
@@ -0,0 +1,228 @@
+
+#### md5
+
+[md5](https://tools.ietf.org/html/rfc1321) is an alternate method to specify hash sum. It's usage is discouraged and `sha256` is preferred.
+
+### Other cases
+
+There are other ways to specify sources to cover other cases.
+
+#### Source code & license
+
+Certain projects provide license on their own, and released artifacts do not include it. In this case, a license URL can be provided separately:
+
+```
+sources:
+  8.0.0:
+    - url: https://github.com/approvals/ApprovalTests.cpp/releases/download/v.8.0.0/ApprovalTests.v.8.0.0.hpp
+      sha256: e16a97081f8582be951d95a9d53dc611f1f5a84e117a477029890d0b34ae99d6
+    - url: "https://raw.githubusercontent.com/approvals/ApprovalTests.cpp/v.8.0.0/LICENSE"
+      sha256: c71d239df91726fc519c6eb72d318ec65820627232b2f796219e87dcf35d0ab4
+```
+
+#### Several source code archives
+
+Some projects may include multiple tarballs as a part of release, [OpenCV](https://opencv.org/) is an example which includes auxiliary [contrib](https://github.com/opencv/opencv_contrib) archive:
+
+```
+sources:
+  "4.5.0":
+    - sha256: dde4bf8d6639a5d3fe34d5515eab4a15669ded609a1d622350c7ff20dace1907
+      url: https://github.com/opencv/opencv/archive/4.5.0.tar.gz
+    - sha256: a65f1f0b98b2c720abbf122c502044d11f427a43212d85d8d2402d7a6339edda
+      url: https://github.com/opencv/opencv_contrib/archive/4.5.0.tar.gz
+```
+
+#### Different source code archives per configuration
+
+This is the most advanced and sophisticated use-case, but no so common. Some projects may provide different sources for different platforms for awkward reasons, it could be expressed as:
+
+```
+sources:
+  "0066":
+    "Macos":
+      "apple-clang":
+        "x86":
+          - url: "https://naif.jpl.nasa.gov/pub/naif/misc/toolkit_N0066/C/MacIntel_OSX_AppleC_32bit/packages/cspice.tar.Z"
+            sha256: "9a4b5f674ea76821c43aa9140829da4091de646ef3ce40fd5be1d09d7c37b6b3"
+        "x86_64":
+          - url: "https://naif.jpl.nasa.gov/pub/naif/misc/toolkit_N0066/C/MacIntel_OSX_AppleC_64bit/packages/cspice.tar.Z"
+            sha256: "f5d48c4b0d558c5d71e8bf6fcdf135b0943210c1ff91f8191dfc447419a6b12e"
+```
+
+This approach requires a special code within [build](https://docs.conan.io/en/latest/reference/conanfile/methods.html#build) method to handle.
+
+## patches
+
+Sometimes sources provided by project require patching for various reasons. The `conandata.yml` file is the right place to indicate this information as well.
+
+This section follows the same pattern as the `sources` above: one entry per version with a list of patches to apply.
+
+```yaml
+patches:
+  "1.2.0":
+    - patch_file: "patches/1.2.0-002-link-core-with-find-library.patch"
+      base_path: "source_subfolder"
+      url: "https://a-url-to-a-pull-request-mail-list-topic-issue-or-question"
+      sha256: "qafe4rq54533qa43esdaq53ewqa5"
+      patch_type: "portability"
+      patch_source: ...
```

Comment:
What is the difference between `url` and `patch_source` fields?

Reply:
`url` is URI for patch file itself (.diff or .patch format)
`patch_source` might be an URL to a JIRA ticker, mail list thread or similar thing
it's explained below I guess

Reply:
Does that mean the example is wrong?

Would a practical example be...

```yaml
url: "https://github.com/conan-io/conan-center-index/pull/11713.diff"
patch_source: "https://github.com/conan-io/conan-center-index/pull/11713"
```

or maybe 

```yaml
url: "https://www.boost.org/patches/1_79_0/0001-json-array-erase-relocate.patch"
patch_source: "https://github.com/boostorg/json/issues/692"
```

It's hard to image another situation where the `url` is different from the `patch_source`

Reply:
yes, example requires a small fix.
in general, `url` and `patch_source` are going to be different in 99% of cases, as `patch_source` points to some discussion or bug report.

Reply:
I think this is too much to ask contributors for...

Reply:
> I think this is too much to ask contributors for...

I agree 🤔